Bump monerod version to v0.18.3.4

colder-is-better opened the pull request.

nmfretz commented:
Thanks for this update @colder-is-better! Sorry for the delay in reviewing this while we had our heads down on the recent umbrelOS beta release.

FYI, the `monerod` service must have a restart policy of "unless-stopped" in order for the daemon to start back up automatically when it is "restarted" from the UI in Advanced Settings. Without this, when a user goes to change their monero config in Advanced Settings and clicks "Save and restart monero node", their node would not actually come back online. I have changed it back here https://github.com/getumbrel/umbrel-apps/pull/1808/commits/74964e88d62f9a7c3e4cc26bf09427509f5ec01d.

I have tested a fresh install and update and everything is working well. I also added some release notes:

<img width="530" alt="image" src="https://github.com/user-attachments/assets/0bc91b0f-97be-46b9-87ca-1b4ef5a3c4e9">

colder-is-better commented:
> Thanks for this update @colder-is-better! Sorry for the delay in reviewing this while we had our heads down on the recent umbrelOS beta release.

Hey, no problem -- that was a quick review on my book!
 
> FYI, the `monerod` service must have a restart policy of "unless-stopped" in order for the daemon to start back up automatically when it is "restarted" from the UI in Advanced Settings. Without this, when a user goes to change their monero config in Advanced Settings and clicks "Save and restart monero node", their node would not actually come back online. I have changed it back here [74964e8](https://github.com/getumbrel/umbrel-apps/commit/74964e88d62f9a7c3e4cc26bf09427509f5ec01d).

I already managed to bump into this issue, and I was scratching my head trying to figure out what happened. Thanks for fixing and clarifying this!

> I have tested a fresh install and update and everything is working well. I also added some release notes:

Very nice. From now on, I'll be sure to add those myself.
